superdesk/storage/desk_media_storage.py

```diff
@@ -63,35 +63,84 @@ def url_for_download(self, media_id, content_type=None):
     def fetch_rendition(self, rendition):
         return self.get(rendition.get('media'), 'upload')
 
-    def put(self, content, filename=None, content_type=None, metadata=None, resource=None, **kwargs):
+    def put(self, content, filename=None, content_type=None, metadata=None, resource=None, folder=None, **kwargs):
         """Store content in gridfs.
 
         :param content: binary stream
         :param filename: unique filename
         :param content_type: mime type
         :param metadata: file metadata
         :param resource: type of resource
+        :param str folder: Folder that the file will be stored in
+        :return str: The ID that was generated for this object
         """
         if '_id' in kwargs:
             kwargs['_id'] = bson.ObjectId(kwargs['_id'])
+
+        if folder:
```

Comment:
we could also use gridfs buckets for folder, that would avoid that regex query in gc, but not sure if it's a big deal